### PR TITLE
feat: add automated release flow

### PR DESCRIPTION
**Pull request by peterdeme**
Add automated release flow

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+            const get_change_log_diff = require('./scripts/get_changelog_diff.js')
+            core.exportVariable('CHANGELOG', get_change_log_diff())
+
+            // Getting the release version from the PR source branch
+            // Source branch looks like this: release-1.0.0
+            const version = context.payload.pull_request.head.ref.split('-')[1]
+            core.exportVariable('VERSION', version)
+
+      - name: Install dependencies
+        run: npm ci
```
Comment (mahboubii) [inline .github/workflows/release.yml:31]: repo uses yarn to track dependencies, this actions takes care of caching/building the sdk
```suggestion
      - uses: ./.github/actions/setup-node
```

File: .github/workflows/initiate_release.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0 # gives the changelog generator access to all previous commits
+
+      - name: Update CHANGELOG.md, package.json and push release branch
+        env:
+          VERSION: ${{ github.event.inputs.version }}
+        run: |
+          npx --yes standard-version@9.3.2 --release-as "$VERSION" --skip.tag --skip.commit --tag-prefix=v
```
Comment (mahboubii) [inline .github/workflows/initiate_release.yml:23]: better to add `standard-version` to devDeps

File: .github/workflows/initiate_release.yml [hunk 3]
```diff
@@ -0,0 +1,47 @@
+name: Create release PR
+
+on:
+  workflow_dispatch:
+    inputs:
+      version:
+        description: "The new version number with 'v' prefix. Example: v1.40.1"
+        required: true
```
Comment (mahboubii) [inline .github/workflows/initiate_release.yml:8]: can we add a check to make sure the input is bigger than prev version?

**Review by mahboubii — COMMENTED** [review]
good stuff 👍

File: .github/workflows/initiate_release.yml (hunk 3 above)
Comment (peterdeme) [reply .github/workflows/initiate_release.yml:8]: ✅

File: .github/workflows/initiate_release.yml (hunk 2 above)
Comment (peterdeme) [reply .github/workflows/initiate_release.yml]: ✅

File: .github/workflows/lint.yml [hunk 4]
```diff
@@ -6,6 +6,12 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
       - uses: ./.github/actions/setup-node
+
+      - uses: wagoid/commitlint-github-action@v4
```
[the remark this answers is not in the document]
Comment (peterdeme) [reply .github/workflows/lint.yml]: how about now @mahboubii? 🙂

File: .github/workflows/release.yml [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+            // Source branch looks like this: release-1.0.0
+            const version = context.payload.pull_request.head.ref.split('-')[1]
+            core.exportVariable('VERSION', version)
+
+      - uses: ./.github/actions/setup-node
+
+      - name: Publish package
+        run: npm publish
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
```
Comment (peterdeme) [inline .github/workflows/release.yml:35]: Can someone add this please?

File: .github/workflows/lint.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (mahboubii) [reply .github/workflows/lint.yml]: looks good, thanks!

File: .github/workflows/release.yml (hunk 5 above)
Comment (ferhatelmas) [reply .github/workflows/release.yml:35]: yup, will ping for it to be added org level

**Review by ferhatelmas — APPROVED** [review]
LGTM

**Comment by ferhatelmas** [comment]
@peterdeme token is there, we can merge. Also, could you remove Amin from codeowners?

**Review by AnatolyRugalev — DISMISSED** [review]

**Review by ferhatelmas — APPROVED** [review]

**Review by AnatolyRugalev — APPROVED** [review]

**Pull request merged**